### PR TITLE
New component: Sumo Logic extension

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:**

Adds new component (sumologicextension). This is only factory and configuration PR

**Link to tracking Issue:** #29601 

**Testing:** N/A

**Documentation:**

- readme
- metadata.yaml

### CODE REVIEW

File: extension/sumologicextension/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Sumo Logic Extension
+
+**Stability level**: Beta
```
Comment (atoulme) [inline extension/sumologicextension/README.md:3]: doesn't match metadata.yaml

File: extension/sumologicextension/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+type: sumologic
+
+status:
+  class: extension
+  stability:
+    development: [extension]
+  distributions: [sumo]
+  codeowners:
+    active: []
```
[the remark this answers is not in the document]
Comment (sumo-drosiek) [reply extension/sumologicextension/metadata.yaml]: Done

File: extension/sumologicextension/README.md (hunk 1 above)
Comment (sumo-drosiek) [reply extension/sumologicextension/README.md]: Fixed

**Comment by sumo-drosiek** [comment]
@atoulme @dashpole Could you review again?

File: extension/sumologicextension/factory.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	DefaultAPIBaseURL = "https://open-collectors.sumologic.com"
+)
+
+// NewFactory creates a factory for Sumo Logic extension.
+func NewFactory() extension.Factory {
+	return extension.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		createExtension,
+		component.StabilityLevelBeta,
```
Comment (atoulme) [inline extension/sumologicextension/factory.go:28]:
```suggestion
		metadata.ExtensionStability,
```

File: extension/sumologicextension/factory.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+const (
+	// The value of extension "type" in configuration.
+	typeStr           = "sumologic"
+	DefaultAPIBaseURL = "https://open-collectors.sumologic.com"
+)
+
+// NewFactory creates a factory for Sumo Logic extension.
+func NewFactory() extension.Factory {
+	return extension.NewFactory(
+		typeStr,
```
Comment (atoulme) [inline extension/sumologicextension/factory.go:25]:
```suggestion
		metadata.Type,
```

File: extension/sumologicextension/factory.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	"context"
+
+	"github.com/cenkalti/backoff/v4"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+)
+
+const (
+	// The value of extension "type" in configuration.
+	typeStr           = "sumologic"
```
Comment (atoulme) [inline extension/sumologicextension/factory.go:18]:
```suggestion
```

File: extension/sumologicextension/factory.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+//go:generate mdatagen metadata.yaml
+
+package sumologicextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension"
+
+import (
+	"context"
+
+	"github.com/cenkalti/backoff/v4"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
```
Comment (atoulme) [inline extension/sumologicextension/factory.go:13]:
```suggestion
	"go.opentelemetry.io/collector/extension"

	“github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension/internal/metadata”
```

File: extension/sumologicextension/README.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@astencel-sumo](https://www.github.com/astencel-sumo), [@sumo-drosiek](https://www.github.com/sumo-drosiek), [@swiatekm-sumo](https://www.github.com/swiatekm-sumo) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector#development
+[sumo]: https://github.com/SumoLogic/sumologic-otel-collector
+<!-- end autogenerated section -->
+
+This extension is to be used as part of Sumo Logic collector in conjunction with
```
Comment (astencel-sumo) [inline extension/sumologicextension/README.md:14]:
```suggestion
This extension is to be used in conjunction with
```

File: extension/sumologicextension/README.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+  and to always reach out to API to register itself. (default: `false`)
+
+  **NOTE**: if clobber is unset (default) then setting this to true will create
+  a new collector (with new unique name) on Sumo UI on every collector start
+  and create a new one upon registration.
+- `ephemeral`: defines whether the collector will be deleted after 12 hours
+  of inactivity (default: `false`)
+- `time_zone`: defines the time zone of the collector. For a list of all possible
+  values, refer to the `TZ database name` or `Time zone abbreviation` columns in
+  https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
```
Comment (astencel-sumo) [inline extension/sumologicextension/README.md:73]:
```suggestion
- `time_zone`: defines the time zone of the collector, for example "Americas/Los_Angeles".
  For a list of all possible values, refer to the `TZ identifier` column in
  https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
```

File: extension/sumologicextension/README.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+[clobber]: https://help.sumologic.com/docs/send-data/installed-collectors/collector-installation-reference/force-collectors-name-clobber/
+
+## Example Config
+
+```yaml
+extensions:
+  sumologic:
+    installation_token: <token>
+    collector_name: my_collector
+    time_zone: CET
```
Comment (astencel-sumo) [inline extension/sumologicextension/README.md:91]:
```suggestion
    time_zone: Europe/Warsaw
```

**Review by andrzej-stencel — APPROVED** [review]

**Comment by sumo-drosiek** [comment]
> This PR was marked stale due to lack of activity. It will be closed in 14 days.


Can we merge it?
cc: @dashpole @atoulme

**Comment by sumo-drosiek** [comment]
cc: @dashpole

**Comment by sumo-drosiek** [comment]
@open-telemetry/collector-contrib-maintainer

**Pull request merged**